internal/services/appservice/linux_function_app_slot_resource.go

```diff
@@ -0,0 +1,787 @@
+		},
+
+		"functions_extension_version": {
+			Type:        pluginsdk.TypeString,
+			Optional:    true,
+			Default:     "~4",
+			Description: "The runtime version associated with the Function App Slot.",
+		},
+
+		"https_only": {
+			Type:        pluginsdk.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Can the Function App Slot only be accessed via HTTPS?",
+		},
+
+		"identity": helpers.IdentitySchema(),
+
+		"site_config": helpers.SiteConfigSchemaLinuxFunctionAppSlot(),
+
+		"tags": tags.Schema(),
+	}
+}
+
+func (r LinuxFunctionAppSlotResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"custom_domain_verification_id": {
+			Type:      pluginsdk.TypeString,
+			Computed:  true,
+			Sensitive: true,
+		},
+
+		"default_hostname": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"kind": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"outbound_ip_addresses": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"outbound_ip_address_list": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"possible_outbound_ip_addresses": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"possible_outbound_ip_address_list": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+
+		"site_credential": helpers.SiteCredentialSchema(),
+	}
+}
+
+func (r LinuxFunctionAppSlotResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var functionAppSlot LinuxFunctionAppSlotModel
+
+			if err := metadata.Decode(&functionAppSlot); err != nil {
+				return err
+			}
+
+			client := metadata.Client.AppService.WebAppsClient
+			aseClient := metadata.Client.AppService.AppServiceEnvironmentClient
+			servicePlanClient := metadata.Client.AppService.ServicePlanClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			id := parse.NewFunctionAppSlotID(subscriptionId, functionAppSlot.ResourceGroup, functionAppSlot.FunctionAppName, functionAppSlot.Name)
+
+			servicePlanId, err := parse.ServicePlanID(functionAppSlot.ServicePlanId)
+			if err != nil {
+				return err
+			}
+
+			servicePlan, err := servicePlanClient.Get(ctx, servicePlanId.ResourceGroup, servicePlanId.ServerfarmName)
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", servicePlanId, err)
+			}
+
+			sendContentSettings := !functionAppSlot.ForceDisableContentShare
+			if planSku := servicePlan.Sku; planSku != nil && planSku.Tier != nil {
+				switch tier := *planSku.Tier; strings.ToLower(tier) {
+				case "dynamic": // Consumption Plan modifications to request
+					sendContentSettings = false
+				case "elastic": // ElasticPremium Plan modifications to request?
```

Comment:
Do we need to do anything for this case?

Reply:
This switch is being used to call out that there are potentially differences in data sent for different plan types, the intention of leaving in those _currently_ without special modifications is to provide readiness / hints about where/how to do so if needed in future. Since this case is empty, it just drops through, so I felt it was low enough cost to provide hints to future development.